### ISSUE TITLE
In RemoteApp the +decorations switch does nothing

### ISSUE THREAD

**Issue by roland65** [FreeRDP/FreeRDP#6702]
In Linux (Ubuntu 20.04 with GNOME desktop), launching a RemoteApp like this:

xfreerdp /log-level:ERROR +decorations /bpp:32 /u:user /p:password  /app:"C:\Windows\notepad.exe" /v:192.168.xx.xx

doesn't show any window manager (WM) decoration.

The expected behaviour is to show WM decorations with +decorations switch and no decorations with -decorations switch.

**Comment by akallabeth** [comment]
sorry, you misinterpret the meaning of the option. it is just a flag sent to the server and it is up to the server to respect or ignore that.

**Comment by roland65** [comment]
Are you sure of that?

Looking at the code, I see in file client/X11/xf_wi,nddow.c that there is a function:

void xf_SetWindowDecorations(xfContext* xfc, Window window, BOOL show)
{
	PropMotifWmHints hints;
	hints.decorations = (show) ? MWM_DECOR_ALL : 0;
	hints.functions = MWM_FUNC_ALL;
	hints.flags = MWM_HINTS_DECORATIONS | MWM_HINTS_FUNCTIONS;
	hints.inputMode = 0;
	hints.status = 0;
	XChangeProperty(xfc->display, window, xfc->_MOTIF_WM_HINTS, xfc->_MOTIF_WM_HINTS, 32,
	                PropModeReplace, (BYTE*)&hints, PROP_MOTIF_WM_HINTS_ELEMENTS);
}

This function sets the X11 client window decorations according to the value of the 'show' boolean, which can in turn be set to 'true' in the .xf_SetWindowFullscreen() function through the xfc->decorations setting.

So it seems to me that it's a client switch, or perhaps I missed something...

**Comment by roland65** [comment]
And I just checked that in desktop mode, the +decorations switch sets X11 client decorations while -decorations removes it,

**Comment by akallabeth** [comment]
@roland65 yes, that is true for desktop sessions, but for rails (remote app) it is quite a different kind of beast.

**Comment by roland65** [comment]
Hi again, I think you're wrong. The +decorations switch is a local option that has nothing to do with the RemoteApp server. In desktop mode, it allows displaying or hiding the decorations of the local client windows. It should be the same in RemoteApp mode.

In source file client/X11/xf_window.c, there is a function;

void xf_SetWindowDecorations(xfContext* xfc, Window window, BOOL show)
{
	PropMotifWmHints hints;
	hints.decorations = (show) ? MWM_DECOR_ALL : 0;
	hints.functions = MWM_FUNC_ALL;
	hints.flags = MWM_HINTS_DECORATIONS | MWM_HINTS_FUNCTIONS;
	hints.inputMode = 0;
	hints.status = 0;
	XChangeProperty(xfc->display, window, xfc->_MOTIF_WM_HINTS, xfc->_MOTIF_WM_HINTS, 32,
	                PropModeReplace, (BYTE*)&hints, PROP_MOTIF_WM_HINTS_ELEMENTS);
}

This function clearly sets the decorations of the local X11 window.

This works in desktop mode and it should work the same in RemoteApp mode.

So please reopen the bug and fix it.

**Comment by akallabeth** [comment]
@roland65 
1. ok, mixed that up with `+aero` et al which are indeed flags for the server
2. The desired option is working (as you already mentioned) for desktop sessions
3. `rails` or `remote app` does not work (properly) with local decorations (or at least needs a major rewrite) if you want that feature. You´re the first to request that, so, pull requests welcome ;)

**Comment by roland65** [comment]
IMHO, the simplest thing to do is to warn the user that the +decorations option doesn't work in RemoteApp mode.

**Comment by DAK85** [comment]
It is a pity that the support for this option has not been implemented. Faced with a situation where the application does not work well without local decorations.
